TeamCode/src/main/java/org/firstinspires/ftc/teamcode/helpers/LoopTimeMonitor.java

```diff
@@ -0,0 +1,46 @@
+package org.firstinspires.ftc.teamcode.helpers;
+import java.util.TreeSet;
+import java.util.Iterator;
+
+
+public class LoopTimeMonitor {
+    private long loopStartTime;
+    private TreeSet<Long> loopTimes = new TreeSet<>();
+
+    public enum ElementSelectionType {
+        TOP_N_ELEMENTS,
+        TOP_PERCENTILE_ELEMENTS
+    }
+    
+    public void loopStart() {
+        this.loopStartTime = System.currentTimeMillis();
+    }
+
+    public void loopEnd() {
+        long loopTime = System.currentTimeMillis() - this.loopStartTime;
+        this.loopTimes.add(loopTime);
```

Comment:
if `loopEnd()` is called when `loopStart()` has not been called, maybe throw an exception instead of crashing because it's null?

Reply:
LGTM 🚀🥳

Reply:
plus, I'm pretty sure that when you declare a long in java, it's immediately assigned a value of zero

Reply:
tada pohui go merge